### DISCUSSION TITLE
source controller deployment strategy

### DISCUSSION THREAD

**Discussion by danielmilanov** [fluxcd/flux2#4064] [General] [answered]
Hi, could someone please shed some light why the deployment strategy on the source controller is set to _recreate_ by default? 

A broken installation can't be fixed by just merging a fix, you have to intervene manually, and a broken source controller hits everything else.

Looking at v2.0.0 

`flux install --components source-controller --export` for reference.

TIA

**Comment by stefanprodan** [comment, marked as answer]
If you don't use a [persistent volume for the artifacts storage](https://fluxcd.io/flux/cheatsheets/bootstrap/#persistent-storage-for-flux-internal-artifacts), you can patch the source-controller deployment strategy type and set it to `RollingUpdate`.

**Reply by danielmilanov** [reply]
Yes, thank you.

**Comment by mo-rieger** [comment]
The `RollingUpdate` strategy prevents updates of the source-controller, as the new pod will never become ready due to failing leader election as long as the old pod is running. See  https://github.com/fluxcd/source-controller/blob/main/main.go#L269
